src/imitation/algorithms/bc.py

```diff
@@ -445,12 +436,13 @@ def _on_epoch_end(epoch_number: int):
             batches_with_stats = tqdm.tqdm(
                 batches_with_stats,
                 unit="batch",
-                total=n_batches,
+                total=n_minibatches,
             )
             tqdm_progress_bar = batches_with_stats
 
-        for (batch_num, batch_size, num_samples_so_far), batch in batches_with_stats:
-            loss = self.trainer(batch)
+        def process_batch():
```

Comment:
This could just be a (private) instance method as it only closes over instance variable `self.optimizer`

Reply:
I'm not sure, it also needs a bunch of local variables. In theory, it can be rewritten as a method, but would require quite a few arguments